lib/utility/include/Logger.h

```diff
@@ -1,28 +1,41 @@
 #pragma once
 
+#include <iostream>
+#include <mutex>
+#include <sstream>
+
+#include "FeatureFlags.h"
+
 namespace Utility {
 
-class Logger {
+class Logger;
+extern Logger logger;
+
+class Logger : public FileHandle {
  public:
-  // C-style printf
-  static int printf(const char* format, ...);
-
-  Logger& operator<<(long n);
-  Logger& operator<<(unsigned long n);
-  Logger& operator<<(bool n);
-  Logger& operator<<(short n);
-  Logger& operator<<(unsigned short n);
-  Logger& operator<<(int n);
-  Logger& operator<<(unsigned int n);
-  Logger& operator<<(long long n);
-  Logger& operator<<(unsigned long long n);
-
-  Logger& operator<<(double f);
-  Logger& operator<<(float f);
-  Logger& operator<<(long double f);
-
-  Logger& operator<<(char c);
-  Logger& operator<<(const char* format);
+  Logger();
+
+  template <typename T>
+  Logger& operator<<(T out) {
+    std::scoped_lock<Mutex> lock(m_mutex);
```

Comment:
just an fyi, this will crash on interrupt

Reply:
good point, I think we shouldn't be printing in an isr anyway tho

Reply:
100% agree for normal operation tho it might be nice to have a deferred print for debugging some isr code(ex. Can). But yeah later problem